core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/common/nearby/BlockDistributionNearbyRandom.java

```diff
@@ -68,4 +68,20 @@ public int getOverallSizeMaximum() {
         return sizeMaximum;
     }
 
+    @Override
+    public boolean equals(Object other) {
+        if (this == other)
+            return true;
+        if (other == null || getClass() != other.getClass())
+            return false;
+        BlockDistributionNearbyRandom that = (BlockDistributionNearbyRandom) other;
+        return sizeMinimum == that.sizeMinimum && sizeMaximum == that.sizeMaximum
+                && Double.compare(that.sizeRatio, sizeRatio) == 0
+                && Double.compare(that.uniformDistributionProbability, uniformDistributionProbability) == 0;
```

Comment:
Why use compare() over equals()?
It it somehow makes sense (like it does for BigDecimal which have the scale complexity), maybe add a short comment for that?

Reply:
It was auto-generated. I'm guessing it was because the fields are `double` and using `Objects.equals(...)` would have boxed them to `Double`.